crates/esplora/src/async_ext.rs

```diff
@@ -226,28 +224,33 @@ async fn chain_update<S: Sleeper>(
     local_tip: &CheckPoint,
     anchors: &BTreeSet<(ConfirmationBlockTime, Txid)>,
 ) -> Result<CheckPoint, Error> {
-    let mut point_of_agreement = None;
+    let mut cp = local_tip.clone();
     let mut conflicts = vec![];
-    for local_cp in local_tip.iter() {
-        let remote_hash = match fetch_block(client, latest_blocks, local_cp.height()).await? {
+
+    let mut tip = loop {
+        let remote_hash = match fetch_block(client, latest_blocks, cp.height()).await? {
             Some(hash) => hash,
             None => continue,
```

Comment:
Is the intention that we continuously try fetch the same block of height until this returns some? 

This will be problematic if the server is very behind as syncing will hang and start spamming the server.